Problems compiling on FreeBSD (again) - sighandler

See https://github.com/GenericMappingTools/gmt/issues/4396 for a similar(?) issue.

```
CMake version: 3.20.4

.. snip ..

-- Configuring done
-- Generating done
-- Build files have been written to: /home/anbj/gmt/b
[  1%] Built target gen_gmt_headers
[  1%] Built target gen_gmt_suppl_supplements_headers
Consolidate compiler generated dependencies of target pslib
[  3%] Built target pslib
Consolidate compiler generated dependencies of target gmtlib
[  3%] Building C object src/CMakeFiles/gmtlib.dir/gmt_common_sighandler.c.o
/home/anbj/gmt/gmt/src/gmt_common_sighandler.c:180:14: error: no member named
      'mc_eip' in 'struct __mcontext'
        array [0] = UC_IP (uc); /* caller's address */
                    ^~~~~~~~~~
/home/anbj/gmt/gmt/src/gmt_common_sighandler.c:77:49: note: expanded from macro
      'UC_IP'
#  define UC_IP(uc) ((void *) (uc)->uc_mcontext.mc_eip)
                              ~~~~~~~~~~~~~~~~~ ^
1 error generated.
gmake[2]: *** [src/CMakeFiles/gmtlib.dir/build.make:197: src/CMakeFiles/gmtlib.dir/gmt_common_sighandler.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:727: src/CMakeFiles/gmtlib.dir/all] Error 2
gmake: *** [Makefile:156: all] Error 2
```

--- a/src/gmt_common_sighandler.c
+++ b/src/gmt_common_sighandler.c
@@ -85,6 +85,8 @@ void backtrace_symbols_fd(void *const *buffer, int size, int fd) {
 #  define UC_IP(uc) ((void *) (uc)->uc_mcontext.mc_rip)
 # elif defined( __arm__)
 #  define UC_IP(uc) ((void *) (uc)->uc_mcontext.arm_pc)
+# elif defined( __aarch64__)
+#  define UC_IP(uc) ((void *) (uc)->uc_mcontext.mc_gpregs.gp_elr)
 # elif defined(__ppc__)
 #  define UC_IP(uc) ((void *) (uc)->uc_mcontext.mc_srr0)
 # else

